core/pkg/otel/metrics.go

```diff
@@ -46,14 +53,29 @@ func (r MetricsRecorder) InFlightRequestEnd(ctx context.Context, attrs []attribu
 	r.httpRequestsInflight.Add(ctx, -1, attrs...)
 }
 
-func (r MetricsRecorder) Impressions(ctx context.Context, key, variant string) {
+func (r MetricsRecorder) Impressions(ctx context.Context, reason, variant, key string) {
 	r.impressions.Add(ctx, 1, []attribute.KeyValue{
+		semconv.FeatureFlagProviderName(FlagdProviderName),
 		semconv.FeatureFlagKey(key),
 		semconv.FeatureFlagVariant(variant),
-		semconv.FeatureFlagProviderName("flagd"),
+		FeatureFlagReason(reason),
+	}...)
+}
+
+func (r MetricsRecorder) Reasons(ctx context.Context, reason string) {
+	r.reasons.Add(ctx, 1, []attribute.KeyValue{
+		semconv.FeatureFlagProviderName(FlagdProviderName),
+		FeatureFlagReason(reason),
```

Comment:
In cases where the `Reason` is `"error", we would also have an error code we could add here. It will just be nil or empty in non-error-reason cases. What do you think @thisthat @beeme1mr ?

Reply:
surely, we can add another dimension describing error type, when the reason is error. But will wait if rest of the team agree

Reply:
I think that error codes would be valuable. 👍 

@thisthat any concerns about this?

Reply:
already adapted, if it will be declined, I will just revert the last commit